### PR TITLE
ci: Release with Github Actions

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Transfer the release from Travis to Github Actions. To achieve this several workflows are merged together into
buildandtest.yml because the release job should only run if the units tests are green, the samples compile and Cocoapods and Carthage are validated. This would have been hard to achieve across several workflows.

## :bulb: Motivation and Context

We want to release this SDK with Github Actions so we can get rid of Travis.

## :green_heart: How did you test it?

With a test release branch: https://github.com/getsentry/sentry-cocoa/runs/1259925396 that uploaded the artifact to zeus https://zeus.ci/gh/getsentry/sentry-cocoa/builds/2416/.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I updated the docs if needed
- [x] No breaking changes

### CODE REVIEW

File: .github/workflows/buildandtest.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+    steps:
+      - uses: actions/checkout@v2
+      - run: ./scripts/ci-select-xcode.sh
+      - name: Running tests
+        run: >-
+          xcodebuild 
+          -workspace Sentry.xcworkspace 
+          -scheme Sentry 
+          -configuration Debug
```
Comment (bruno-garcia) [inline .github/workflows/buildandtest.yml:42]: Is it possible to test in release more? That's the type that gets shipped so always good to run CI tests on that if the platform supports it.

**Review by bruno-garcia — APPROVED** [review]
noice

**Comment by bruno-garcia** [comment]
once we release and make sure the zeus stuff work, I'll copy over to other repos :p

Maybe worth a github action eh?

**Comment by philipphofmann** [comment]
> once we release and make sure the zeus stuff work, I'll copy over to other repos 

I actually got that from react-native: https://github.com/getsentry/sentry-react-native/blob/594a13770ccfeac30e30e7476e11ce3817db7004/.github/workflows/buildandtest.yml#L134-L141

> Maybe worth a github action eh?

Isn't somebody already working on this?

**Comment by bruno-garcia** [comment]
> Isn't somebody already working on this?


Not that I know of

**Pull request merged**